app/src/main/java/com/example/atheneum/viewmodels/FirebaseRefUtils/DatabaseWriteHelper.java

```diff
@@ -398,6 +399,57 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
         });
     }
 
+    public static void makeAllNotificationsSeen(String userID) {
+        Query notificationsRef = NotificationsRefUtils.getNotificationsRef(userID);
+        notificationsRef.addChildEventListener(new ChildEventListener() {

```

Comment:
Note: this child event listener is always active so you can potentially end up making notifications that come in the future be seen. See my PR #201 for why that can happen (me and Alex noticed it happening for requests where they randomly auto-decline when you try to re-request a book you just read) and how to fix that. You should use a single value event listener since you are intending to only delete the current set of notifications not all future ones.

Reply:
Fixed. Other than this, I can't reproduce the bug you had.